src/app/Http/Controllers/Operations/FetchOperation.php

```diff
@@ -60,13 +60,13 @@ private function fetch($arg)
         $config['searchable_attributes'] = $config['searchable_attributes'] ?? $model_instance->identifiableAttribute();
         $config['query'] = isset($config['query']) && is_callable($config['query']) ? $config['query']($config['model']) : $model_instance; // if a closure that has been passed as "query", use the closure - otherwise use the model
 
-        // FetchOperation is aware of an optional parameter 'keys' that will fetch you the entity/entities that match the provided keys
-        if (request()->has('keys')) {
-            $decoded_keys = json_decode(request()->get('keys'));
-            if (is_array($decoded_keys)) {
-                return $model_instance->whereIn($model_instance->getKeyName(), $decoded_keys)->get();
+        // FetchOperation is aware of an optional parameter 'related_keys' that will fetch you the entity/entities that match the provided keys
+        if (request()->has('related_keys')) {
+            $array_keys = explode(',', request()->get('related_keys'));
+            if (is_array($array_keys) && count($array_keys) > 1) {
+                return $model_instance->whereIn($model_instance->getKeyName(), $array_keys)->get();
             } else {
-                return $model_instance->where($model_instance->getKeyName(), $decoded_keys)->first();
+                return $model_instance->where($model_instance->getKeyName(), $array_keys)->get();
```

Comment:
`$model_instance->find($array_keys)` would probably be shorter.

Reply:
read my previous concern please in findMany :)